src/main/java/com/github/stickerifier/stickerify/media/MediaHelper.java

```diff
@@ -273,6 +286,22 @@ private static File createPngFile(BufferedImage image) throws TelegramApiExcepti
 		return pngImage;
 	}
 
+	/**
+	 * Performs an optimization aimed to reduce the image's size using {@link PngOptimizer}.
+	 *
+	 * @param pngImage the file to optimize
+	 * @throws IOException if the optimization process fails
+	 * @throws TelegramApiException if the image size could not be reduced enough to meet Telegram's requirements
+	 */
+	private static void optimizeImage(File pngImage) throws IOException, TelegramApiException {
```

Comment:
This should be MediaOptimizationException

Reply:
Nope because `isFileSizeLowerThan` throws `TelegramApiException`